felix/fv/infrastructure/infra_k8s.go

```diff
@@ -805,6 +807,37 @@ func (kds *K8sDatastoreInfra) AddWorkload(wep *libapi.WorkloadEndpoint) (*libapi
 	if wep.Labels != nil {
 		podIn.ObjectMeta.Labels = wep.Labels
 	}
+	if wep.Spec.QoSControls != nil {
+		if podIn.Annotations == nil {
+			podIn.Annotations = map[string]string{}
+		}
+		if wep.Spec.QoSControls.IngressBandwidth != 0 {
+			podIn.Annotations[conversion.AnnotationQoSIngressBandwidth] = resource.NewQuantity(wep.Spec.QoSControls.IngressBandwidth, resource.DecimalSI).String()
+		} else {
+			delete(podIn.Annotations, conversion.AnnotationQoSIngressBandwidth)
+		}
+		if wep.Spec.QoSControls.IngressBurst != 0 {
+			podIn.Annotations[conversion.AnnotationQoSIngressBurst] = resource.NewQuantity(wep.Spec.QoSControls.IngressBurst, resource.DecimalSI).String()
+		} else {
+			delete(podIn.Annotations, conversion.AnnotationQoSIngressBurst)
+		}
+		if wep.Spec.QoSControls.EgressBandwidth != 0 {
+			podIn.Annotations[conversion.AnnotationQoSEgressBandwidth] = resource.NewQuantity(wep.Spec.QoSControls.EgressBandwidth, resource.DecimalSI).String()
+		} else {
+			delete(podIn.Annotations, conversion.AnnotationQoSEgressBandwidth)
+		}
+		if wep.Spec.QoSControls.EgressBurst != 0 {
+			podIn.Annotations[conversion.AnnotationQoSEgressBurst] = resource.NewQuantity(wep.Spec.QoSControls.EgressBurst, resource.DecimalSI).String()
+		} else {
+			delete(podIn.Annotations, conversion.AnnotationQoSEgressBurst)
+		}
+
+	} else if podIn.Annotations != nil {
+		delete(podIn.Annotations, conversion.AnnotationQoSIngressBandwidth)
+		delete(podIn.Annotations, conversion.AnnotationQoSIngressBurst)
+		delete(podIn.Annotations, conversion.AnnotationQoSEgressBandwidth)
+		delete(podIn.Annotations, conversion.AnnotationQoSEgressBurst)
+	}
```

Comment:
Do we need code here to set `podIn.Annotations` back to nil if the map is now empty?